dora-parser/src/ast/dump.rs

```diff
@@ -297,6 +298,42 @@ impl<'a> AstDumper<'a> {
         });
     }
 
+    fn dump_annotation(&mut self, annotation: &Annotation) {
+        dump!(
+            self,
+            "annotation {} @ {} {}",
+            self.str(annotation.name),
+            annotation.pos,
+            annotation.id
+        );
+
+        self.indent(|d| {
+            dump!(d, "params");
+            if let Some(params) = &annotation.term_params {
+                for param in params {
+                    d.dump_annotation_param(param);
+                }
+            }
+        });
+    }
+
+    fn dump_annotation_param(&mut self, param: &AnnotationParam) {
+        dump!(self, "param {} @ {}", self.str(param.name), param.pos);
+
+        self.indent(|d| d.dump_type(&param.data_type));
+    }
+
+    fn dump_annotation_usages(&self, annotation_usages: &AnnotationUsages) {
```

Comment:
please fix this warning

Reply:
Added a `#[allow(dead_code)]` until it is used, not sure why it isn't showing up here.